starwars_api/models.py

```diff
@@ -53,19 +61,46 @@ def __repr__(self):
 
 
 class BaseQuerySet(object):
-
+    RESOURCE_NAME = None
+    
     def __init__(self):
-        pass
-
+        self.page_index = 0
+        self.total_index = 1
+        self.results = None
+        self.total_count = None
+        self.next_page_number = 1
+        self.make_request()
+
+    def make_request(self):
+        param = "?page={}".format(self.next_page_number)
+        get_method_name = "get_{resource_name}".format(resource_name = self.RESOURCE_NAME) 
+        self.data = getattr(api_client, get_method_name)(param)
+        self.total_count = self.data['count']
+        self.results = self.data['results']
+        self.next_page_number += 1
+        
     def __iter__(self):
-        pass
+        return self
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
+        if self.results is None:
+            self.make_request()
+        if self.total_index > self.total_count:
+            raise StopIteration
+        if self.page_index == 10:
```

Comment:
The iterable code looks really great, the only suggestion I would have is that instead of hardcoding in a limit of 10 per page, you could just try iterating over self.results until you raise an IndexError.

Great job overall on the code!